index.html

```diff
@@ -49,19 +49,19 @@ <h1 class="large l-align-center l-mb-mega">&lt;gaia-button&gt;</h1>
   <hr/>
 
   <section>
-    <gaia-button><i data-icon="back-light" aria-hidden="true"></i><span>Back</span></gaia-button>
-    <gaia-button><span>Forward</span><i data-icon="forward-light" aria-hidden="true"></i></gaia-button>
+    <gaia-button><span data-l10n-id="back" data-icon="back-light">Back</span></gaia-button>
+    <gaia-button><span data-l10n-id="forward" data-icon-end="forward-light">Forward</span></gaia-button>
   </section>
 
   <hr/>
 
   <section flex>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="bug"></i></gaia-button>
-    <gaia-button circular><i data-icon="video"></i></gaia-button>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="reload"></i></gaia-button>
-    <gaia-button circular><i data-icon="airplane"></i></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="bug" data-icon="bug"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="video" data-icon="video"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="reload" data-icon="reload"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="airplane" data-icon="airplane"/></gaia-button>
```

Comment:
Span is irrelevant to accessibility here. The aria-label that is applied with data-l10n-id on gaia-button itself describing the button completely should be all that's necessary from a11y standpoint.